test/testdata/lsp/code_actions/extract_class_func.rb

```diff
@@ -0,0 +1,36 @@
+# typed: strict
+# selective-apply-code-action: refactor.extract
+
+module Foo
+  extend T::Sig
+
+  sig {returns(String)}
+  def self.greeting
+         # ^^^^^^^^ apply-code-action: [A] Extract method to module
+    'Hello'
+  end
+  
+  sig do
+    params(x: String)
+    .returns(String)
+  end
+  def name(x)
+    "#{Foo.greeting} #{x}"
+  end
+end
+
+module A
+  class B
+    extend T::Sig
+
+    sig {void}
+    def bar
+      m = Foo
+      Foo.greeting
```

Comment:
How does this handle when `m` comes from the sig:

```
sig {params(m: T.class_of(Foo)).void}
def example3(m)
  m.greeting
end
```

? If you don't mind adding a test for this as well, that'd be great.